resolwe_bio/processes/import_data/seq_reads.yml

```diff
@@ -218,3 +218,248 @@
+
+        elif [ "${ENCODING}" != "Sanger / Illumina 1.9" ]; then
+          re-error "Only Sanger / Illumina 1.9 / llumina 1.5 / Illumina 1.3 encoding is supported."
+        fi
+      done
+
+      FASTQGZ=(*.fastq.gz)
+      re-save-file-list fastq "${FASTQGZ[@]}"
+
+      FASTQC=(*_fastqc.zip)
+      FASTQC_URL=()
+
+      for fqc in "${FASTQC[@]}"
+      do
+        NAME=`basename "$fqc" _fastqc.zip`
+        FASTQC_URL+=("{\"file\":\"fastqc/${NAME}_fastqc/fastqc_report.html\",\"refs\":[\"fastqc/${NAME}_fastqc\"]}")
+      done
+
+      re-save-file-list fastqc_archive "${FASTQC[@]}"
+      re-save-list fastqc_url "${FASTQC_URL[@]}"
+
+- slug: basespace-fastq-paired
+  name: BaseSpace Reads (paired-end)
+  requirements:
+    expression-engine: jinja
+    executor:
+      docker:
+        image: resolwebio/legacy:latest
+    resources:
+      network: true
+      secrets: true
+  data_name: 'BaseSpace {{ upstream_file_ids.0|default("?") }}'
+  version: 0.0.1
+  type: data:reads:fastq:paired
+  category: upload
+  flow_collection: sample
+  persistence: RAW
+  description: >
+    Import paired-end reads in FASTQ format from BaseSpace.
+  input:
+    - name: upstream_file_ids
+      label: BaseSpace NGS upstream read (FASTQ) file IDs
+      type: list:basic:string
+    - name: downstream_file_ids
+      label: BaseSpace NGS downstream read (FASTQ) file IDs
+      type: list:basic:string
+    - name: access_token_secret
+      label: BaseSpace access token
+      type: basic:secret
+      description: >
+        BaseSpace access token secret handle needed to download the files.
+  output:
+    - name: fastq
+      label: Reads file (mate 1)
+      type: list:basic:file
+    - name: fastq2
+      label: Reads file (mate 2)
+      type: list:basic:file
+    - name: fastqc_url
+      label: Quality control with FastQC (Upstream)
+      type: list:basic:file:html
+    - name: fastqc_url2
+      label: Quality control with FastQC (Downstream)
+      type: list:basic:file:html
+    - name: fastqc_archive
+      label: Download FastQC archive (Upstream)
+      type: list:basic:file
+    - name: fastqc_archive2
+      label: Download FastQC archive (Downstream)
+      type: list:basic:file
+  run:
+    runtime: polyglot
+    language: bash
+    program: |
+
+      function extract_filename_from_output() {
+        echo $1 | sed -nE 's/^filename=(.+)/\1/p'
+      }
+
+      SECRET_PATH=/secrets/{{ access_token_secret.handle }}
+
+      FW_FASTQ=()
+      FW_FASTQGZ=()
+      FW_FASTQC=()
+      FW_FASTQC_URL=()
+
+      RW_FASTQ=()
+      RW_FASTQGZ=()
+      RW_FASTQC=()
+      RW_FASTQC_URL=()
+
+      {% for file_id in upstream_file_ids %}
+        BASESPACE_OUTPUT=`basespace_download.py --file-id {{file_id}} --access-token-secret-path ${SECRET_PATH}`
+        re-checkrc "Failed downloading files from BaseSpace."
+
+        GZ_FILE=`extract_filename_from_output ${BASESPACE_OUTPUT}`
+        gzip -dk ${GZ_FILE}
```

Comment:
Filenames can contain spaces. Consistently use quotation marks.

Reply:
Fixed [here](https://github.com/genialis/resolwe-bio/pull/466/files#diff-e860ea00e9e0a400a0646e46faef1896R391), [here](https://github.com/genialis/resolwe-bio/pull/466/files#diff-e860ea00e9e0a400a0646e46faef1896R393), [here](https://github.com/genialis/resolwe-bio/pull/466/files#diff-e860ea00e9e0a400a0646e46faef1896R406) and [here](https://github.com/genialis/resolwe-bio/pull/466/files#diff-e860ea00e9e0a400a0646e46faef1896R408).